eppoclient/eval.go

```diff
@@ -0,0 +1,140 @@
+package eppoclient
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+func (flag flagConfiguration) verifyType(ty variationType) error {
+	if flag.VariationType == ty {
+		return nil
+	} else {
+		return fmt.Errorf("unexpected variation type (expected: %v, actual: %v)", ty, flag.VariationType)
+	}
+}
+
+func (flag flagConfiguration) eval(subjectKey string, subjectAttributes SubjectAttributes) (interface{}, *AssignmentEvent, error) {
+	if !flag.Enabled {
+		return nil, nil, errors.New("the experiment or flag is not enabled")
+	}
+
+	now := time.Now()
+	augmentedSubjectAttributes := augmentWithSubjectKey(subjectAttributes, subjectKey)
+
+	totalShards := flag.TotalShards
+	if totalShards == 0 {
+		totalShards = 10000
+	}
```

Comment:
please return an error instead

Reply:
All other SDKs have `totalShards` field to be optional with `10000` as default value. Go doesn't have a good/quick way to set defaults during parsing, so it's handled here

There are a couple of options:
- If server always sets `totalShards`, we can remove this check or error out. And should make the field non-optional in other SDKs.
- I can try pushing this defaulting to JSON-parsing code
- Leave it as is

Do you have a preference?

Reply:
@schmit what do you want the behavior to do here?

Reply:
I don't have strong preferences here; if this is the way to handle defaults in Go, then maybe stick with this; on the other hand, we can expect the server to always return totalShards and throw an error too.

Reply:
If we decide to remove defaulting or error out instead, I would prefer making this change across all SDKs.

For now I have added a code comment to highlight that this piece is handling defaulting so it's less confusing

Reply:
> All other SDKs have totalShards field to be optional with 10000 as default value.

That is a mistake in the mapping of the API interface (posted below), it is a required field. I have it required in swift (https://github.com/Eppo-exp/eppo-ios-sdk/pull/21/files#diff-62340b699c21c5f22a4be4ed400d736c96b7508d46998f4b361fa1b3a7cf5d6aR97)

<img width="1101" alt="Screenshot 2024-06-06 at 10 57 17 AM" src="https://github.com/Eppo-exp/golang-sdk/assets/57361/a37260fc-cf3c-4eff-9733-89618ca5bb61">

IMO default to `10k` could lead to undesirable assignment behavior - if we accidently pass `0` here from the server it will lead to no assignments which feels like a better outcome then unexpected assignments.

```suggestion
```